signer/tests/integration/contracts.rs

```diff
@@ -1,3 +1,128 @@
+use signer::stacks::api::SubmitTxResponse;
+use tokio::sync::OnceCell;
+
+use secp256k1::ecdsa::RecoverableSignature;
+use secp256k1::Keypair;
+use signer::config::StacksSettings;
+use signer::stacks::api::StacksClient;
+use signer::stacks::wallet::sign_ecdsa;
+use signer::stacks::wallet::MultisigTx;
+use signer::stacks::wallet::SignerStxState;
+use signer::testing::wallet;
+use signer::testing::wallet::AsContractDeploy;
+use signer::testing::wallet::ContractDeploy;
+
+const BOOTSTRAP: &str = include_str!("../../../contracts/contracts/sbtc-bootstrap-signers.clar");
+const DEPOSIT: &str = include_str!("../../../contracts/contracts/sbtc-deposit.clar");
+const REGISTRY: &str = include_str!("../../../contracts/contracts/sbtc-registry.clar");
+const TOKEN: &str = include_str!("../../../contracts/contracts/sbtc-token.clar");
+const WITHDRAWAL: &str = include_str!("../../../contracts/contracts/sbtc-withdrawal.clar");
+
+const TX_FEE: u64 = 15000000;
+
+pub struct SbtcTokenContract;
+
+impl AsContractDeploy for SbtcTokenContract {
+    const CONTRACT_NAME: &'static str = "sbtc-token";
+    const CONTRACT_BODY: &'static str = TOKEN;
+}
+
+pub struct SbtcRegistryContract;
+
+impl AsContractDeploy for SbtcRegistryContract {
+    const CONTRACT_NAME: &'static str = "sbtc-registry";
+    const CONTRACT_BODY: &'static str = REGISTRY;
+}
+
+pub struct SbtcDepositContract;
+
+impl AsContractDeploy for SbtcDepositContract {
+    const CONTRACT_NAME: &'static str = "sbtc-deposit";
+    const CONTRACT_BODY: &'static str = DEPOSIT;
+}
+
+pub struct SbtcWithdrawalContract;
+
+impl AsContractDeploy for SbtcWithdrawalContract {
+    const CONTRACT_NAME: &'static str = "sbtc-withdrawal";
+    const CONTRACT_BODY: &'static str = WITHDRAWAL;
+}
+
+pub struct SbtcBootstrapContract;
+
+impl AsContractDeploy for SbtcBootstrapContract {
+    const CONTRACT_NAME: &'static str = "sbtc-bootstrap-signers";
+    const CONTRACT_BODY: &'static str = BOOTSTRAP;
+}
+
+pub struct SignerKeyState {
+    pub state: SignerStxState,
+    pub keys: [Keypair; 3],
+}
+
+/// Deploy an sBTC smart contract to the stacks node
+async fn deploy_smart_contract<T>(state: &SignerKeyState, client: &StacksClient, deploy: T)
+where
+    T: AsContractDeploy,
+{
+    let mut unsigned = MultisigTx::new_tx(ContractDeploy(deploy), &state.state, TX_FEE);
+
+    let signatures: Vec<RecoverableSignature> = state
+        .keys
+        .iter()
+        .map(|kp| sign_ecdsa(unsigned.tx(), &kp.secret_key()))
```

Comment:
Nit: I find it easier to read and navigate code when we don't import functions.

```suggestion
        .map(|kp| wallet::sign_ecdsa(unsigned.tx(), &kp.secret_key()))
```

Reply:
Oh right, will update.